homeassistant/components/flux_led/light.py

```diff
@@ -21,24 +23,47 @@
     SUPPORT_WHITE_VALUE,
     LightEntity,
 )
-from homeassistant.const import ATTR_MODE, CONF_DEVICES, CONF_NAME, CONF_PROTOCOL
+from homeassistant.const import (
+    ATTR_IDENTIFIERS,
+    ATTR_MANUFACTURER,
+    ATTR_MODE,
+    ATTR_MODEL,
+    ATTR_NAME,
+    CONF_DEVICES,
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PROTOCOL,
+)
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers import entity_platform
 import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_registry import async_entries_for_device
+from homeassistant.util import Throttle
 import homeassistant.util.color as color_util
 
+from .const import (
+    CONF_AUTOMATIC_ADD,
+    CONF_EFFECT_SPEED,
+    DEFAULT_EFFECT_SPEED,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    SIGNAL_ADD_DEVICE,
+    SIGNAL_REMOVE_DEVICE,
+)
+
 _LOGGER = logging.getLogger(__name__)
 
-CONF_AUTOMATIC_ADD = "automatic_add"
-CONF_CUSTOM_EFFECT = "custom_effect"
 CONF_COLORS = "colors"
 CONF_SPEED_PCT = "speed_pct"
 CONF_TRANSITION = "transition"
-
-DOMAIN = "flux_led"
+CONF_CUSTOM_EFFECT = "custom_effect"
 
 SUPPORT_FLUX_LED = SUPPORT_BRIGHTNESS | SUPPORT_EFFECT | SUPPORT_COLOR
-
 MODE_RGB = "rgb"
 MODE_RGBW = "rgbw"
+MODE_RGBCW = "rgbcw"
+MODE_RGBWW = "rgbww"
```

Comment:
Why do we add these in this PR?

Reply:
This PR was part of a larger PR that has been cut up. I tried to cut it down to be just config-related and get it back to the original code as much as possible.  Getting it all the way back to the original and just adding the config-related code will take a lot of work.

If that's what's needed I can do that but, keep in mind that the current integration in production barely functions.  This code keeps it function until it can be completely be updated.

Reply:
Let's keep this PR to just the config flow. It will go much faster if there is less to review.

  
Additional fixes should come in a followup.

Reply:
> Let's keep this PR to just the config flow. It will go much faster if there is less to review.
> 
> Additional fixes should come in a followup.

To be clear.  Do I take this PR back to draft and reduce the code further so that it's just the config flow?

Reply:
The review is already well underway but has mostly focused on the config flow until now.

If you push a commit which reverts changes to `homeassistant/components/flux_led/light.py` which are not required by the config flow implementation, that makes the review much easier, and reverting that commit can be the starting point for the next PR 👍